src/core/selinux-access.c

```diff
@@ -163,6 +164,26 @@ static int access_init(sd_bus_error *error) {
         return 1;
 }
 
+int mac_selinux_callback_check(
+                void *blob,
+                const char *name) {
+
+        _cleanup_free_ char *path = NULL;
+        struct mac_selinux_callback_blob *t = blob;
+        int exists;
+
+        assert(name);
+        assert(t);
+
+        path = path_make_absolute(name, MAC_SELINUX_LABEL_BASE_PATH);
+        if (!path)
+                return -ENOMEM;
```

Comment:
this function really should take an absolute path, and not make the argument absolute on its own. unit files may be located at arbitrary places, not just a single dir

Reply:
for some `unit_file_*` functions it is hard to get the path; also the path might be a user defined instance (e.g. `/etc/systemd/system/ssh.service`) or even `dev/null`. But we need a path, which has a meaningful SELinux file context, like `/lib/systemd/ssh.service`.

Reply:
well, we either have an absolute path or we have an fd. In the latter case we can use fd_get_path() to get a workable path

Reply:
acquiring the source context has bee reworked